### DISCUSSION TITLE
Just a target reached

### DISCUSSION THREAD

**Discussion by agalliazzo** [openthread/openthread#9790] [General]
Hi Guys,
I only want to share a target I reached with OT. After having trouble due to bugs in implementation, by the uP manufacturer I switched to other uP and my thread network seems to works! 

**Really, this was possible also thanks to the support from the OT comunity guys...**

### General info

This is my house openthread based domotic system running with NRF52840, a VM running OTBR with a NRF52840 usb stick as RCP and Openhab as orchestrator.

This is not a simulated environment, devices are inside concrete walls in a 3 floor house with about 12x11m size, some walls, built back in the 1800, are made of stone too...

The whole system is based on CoAP, every device can be pinged from the LAN and controlled by direct connection. All nodes register their hostname using SRP and are discoverable in the LAN. Nodes are also connected to the internet, behind a firewall of course, so they are able to do OTA directly from cloud.

### The network
All SED are temperature, humidity pressure and light sensor transmitting environmental values to main server every 5 min powered by a CR2032 battery that seems to be lasting for at least 1y.

Some REEDs (the one starting with OptoIO) is IO board with 8+8 channel, 8 digital inputs (opto insulated) and 8 bistable relays.  Other REEDs are battery capable pushbutton board now operating on a 12Vdc bus (I'm replacing my old home automation system and wiring was already inside my walls)
 
I have 25 nodes, with about 17 nodes that are REED, 8 are SED. All REED in at the beginning was routers, then some start to downgrade to childs (probably due to lack of secondary connections and childs a router by its own is quite useless). 

On the image below blue pentagon are REEDs acting as routers, the orange one is the OTBR, gray circle are SED or childs. Every link you see is with LQI >= 2, I filtered links with 1 to have a more clean image.

![image](https://github.com/openthread/openthread/assets/2827298/f8d426df-515c-46ef-9450-916f401a6585)

### Performances

Interesting performance fact: pinging from BR to optoiotaverna.local that is the longest route take an AVG trip of 52ms with at least 3 bounces.
![image](https://github.com/openthread/openthread/assets/2827298/affaa3c6-f1d0-43d0-a3b9-68b535e5bcf0)

Please feel free to ask anything if interested :)

**Comment by jwhui** [comment]
@agalliazzo , very cool! Awesome to see your network running well with great performance. Thanks for sharing!
